### DISCUSSION TITLE
Is it possible to get IDE features like autocomplete etc in vs code with solargraph or ruby_lsp?

### DISCUSSION THREAD

**Discussion by steckoverflow** [jeremyevans/sequel#2165] [General]
Hi, so first of all I just wanna say I'm a ruby newbie.

I'm trying to get IDE like editor support with things like autocomplete, go to definition etc for this library. But my experience is vs code is working quite poorly.

Ruby version 3.3.1
Installed via chruby

Vs code extensions
- Ruby v.0.1.8
- Ruby LSP v.0.5.21
- Ruby Solargraph  v.0.24.1

![image](https://github.com/jeremyevans/sequel/assets/66014422/5dda055d-55af-4eb4-b47b-4dfc75e333c8)

![image](https://github.com/jeremyevans/sequel/assets/66014422/28ac6b5a-271b-44c9-a552-47db8dcafe08)

If I type something along the line 

`Ruby
DB = Sequel.postgres('database_name',
       user:'user', 
       password: 'password',
       host: 'host'
       port: 5432, 
       max_connections: 10)`

It runs but I dont get any auto complete or editor support. is it possible or what am I missing?

Again, excuse my ignorance just trying to learn

**Comment by jeremyevans** [comment]
Unfortunately, not being a VS Code user, I don't know the answer to this, so hopefully someone else will have a more helpful answer.  At least IRB can autocomplete `Sequel` and `Sequel.postgres`.  I'm not sure whether the options will be autocompleted by anything, since Sequel uses option hashes and not keyword arguments.

**Reply by steckoverflow** [reply]
I see, but I'm still baffled how I can get better autocomplete in irb than in vs code with all my extensions, I feel like im doing something wrong 
![image](https://github.com/jeremyevans/sequel/assets/66014422/0169882c-bf72-4a5e-9d73-887067e8a7ad)

**Reply by rvyhvn** [reply]
Since Sequel doesn't have YARD docs, I think it's impossible to get the completions/suggestions of this library especially if you use Solargraph or Ruby-LSP from Shopify. I also struggle with this but the owner of the project refuses to add YARD docs. I think the owner have to reconsider to use YARD docs in this good library so new Ruby user can love the language more ! <3
@jeremyevans 

EDIT: Actually, you can generate the YARD docs manually by running `yard gems` so you'll get the completions. Or if you are using a local bundle folder, you can run `bundle exec yard gems`

**Reply by janko** [reply]
> I see, but I'm still baffled how I can get better autocomplete in irb than in vs code with all my extensions, I feel like im doing something wrong

IRB actually runs your Ruby code, so it can dynamically ask the current object for its methods to autocomplete. Ruby LSP cannot do that for code in the text editor, because that code isn't actually evaluated, it's only parsed. AFAIK Ruby LSP only does static analysis, which can only take you so far in terms of autocompletion.
